### DISCUSSION TITLE
Unable to use BelongsTo::relatableQueryUsing on resource created inline from another resource.

### DISCUSSION THREAD

**Discussion by davidbankes** [laravel/nova-issues#6237] [Ideas & Feature Requests]
- Laravel Version: 10.41.0
- Nova Version: 4.32.11
- PHP Version: 8.2
- Database Driver & Version: MySQL 8
- Operating System and Version: Mac OS 14.2.1
- Browser type and version:FireFox 121.0.1
- github repo for bug reproduction: https://github.com/davidbankes/nova-inline-creation

### Description:
Can't use BelongsTo->dependsOn or ->relatableQueryUsing on a resource that can be created via an inline creation button from another resource

### Detailed steps to reproduce the issue on a fresh Nova installation:

Models & relations diagram:

User
->belongsToMany Proposals
->hasMany Documents

Proposal
->belongsToMany Users
->hasMany Requirements

Requirement
->belongsTo Proposal
->belongsTo Document **(inline creation enabled)**

Document
->belongsTo User
->hasOne Requirement

When I'm in a Proposal->Requirement resource, I have a select dropdown to associate a document with it, or I can create a document inline. If I create it inline, I want to be able to only select one of the users associated with the Proposal that the requirement I'm editing belongs to. Based on https://nova.laravel.com/docs/4.0/resources/relationships.html#relatable-query-filtering, I should be able to using a dependent field to do this... make the user field dependent on the requirement field, and then only allow users associated with the proposal whose requirement I'm editing, but I can't make it work.

Also, the documentation linked to above seems pretty much identical to https://nova.laravel.com/docs/resources/fields.html#accessing-request-resource-ids. If I could access the resource ID of the requirement, I'd be home free, but I can't

The URL used for getting the creation fields for the inline creation modal is `http://example.test/nova-api/documents/creation-fields?editing=true&editMode=create&inline=true&viaResource=&viaResourceId=&viaRelationship=`. Shouldn't this include the relevant relationship information? (Understandably, if I was creating a new Requirement, then there'd be no ID to pass, and that's a different problem, but since I'm editing an existing Requirement, shouldn't that information be passed to the creation-fields URL?

**Comment by crynobone** [comment]
Please provide full reproducing repository based on fresh installation as suggested in the bug report template (or you can refer to https://github.com/nova-issues for example)

**Comment by davidbankes** [comment]
@crynobone 

Added: https://github.com/davidbankes/nova-inline-creation

There's an sql dump in the storage folder, too.

I've kept it pretty simple. There's one proposal that two users are collaborating on. The proposal has a requirement, and I want to inline create a document on that requirement. (A document is a file that belongs to a user). I only want the available users to the the users collaborating on the proposal the requirement belongs to. But Nova doesn't give me the information I need to build that filter. I can inline create the document if I don't filter the users, but I want the administrator working on the proposal to only be able to go through the users associated with the proposal.

If you use the SQL dump to get started, the password for all users is `asdf1234`. davidbankes@gmail.com is the user with nova access.

**Comment by nickstewart95** [comment]
Any progress here? I'm interested in this and was running into the same issue

**Comment by davidbankes** [comment]
@crynobone Is there more information you need to evaluate this? I think the "Needs more info" label is incorrect. If there's something more you need, let me know and I'll get it to you.

**Comment by davidbankes** [comment]
@crynobone 

This also applies to a BelongsToMany resource. I can get the button to show up, and theoretically I should be able to filter with `public static relatableEntity()` on the appropriate resource, but the creation-fields request for that modal also lacks the appropriate information.

**Comment by crynobone** [comment]
That's available at the moment is according to the expected behavior, moving this to Idea & Feature Request so we can look into this again in the future.

**Reply by davidbankes** [reply]
Just to be clear, the current expected behavior where:

1. I am creating a new inline creation
2. I cannot filter any field in the creation modal (e.g., a belongsTo) by any information from the parent relation

That is the expected behavior?
